src/panels/lovelace/cards/hui-todo-list-card.ts

```diff
@@ -170,12 +173,36 @@ export class HuiTodoListCard extends LitElement implements LovelaceCard {
     (items?: TodoItem[], sort?: string | undefined): TodoItem[] =>
       items
         ? this._sortItems(
-            items.filter((item) => item.status === TodoItemStatus.NeedsAction),
+            this._filterItemsByDueDate(
+              items.filter(
+                (item) => item.status === TodoItemStatus.NeedsAction
+              ),
+              this._config?.days_to_show
+            ),
             sort
           )
         : []
   );
 
+  private _filterItemsByDueDate(
+    items: TodoItem[],
+    daysToShow?: number | undefined
+  ): TodoItem[] {
+    const ignoreEndDate = daysToShow === undefined || daysToShow === 0;
```

Comment:
Given that items may be past-due, I'm not sure if it makes sense to treat 0 as infinity, or disallow negative values.

`days: 0` could be "show all past due items", or `days: -7` could be "show all items that are overdue by one week or more".

Reply:
So what should be the default?
-7? Or even -30?
Not sure what is best. In my case I don’t really care. I think it should be similar to other cards. Not sure how they are managing this though.

Reply:
I assumed default is undefined/no-filtering.

Reply:
Yes. Ah, I think I misunderstood. You meant adding negative values?

Reply:
I was making two points: 

1) You special cased `0` and made it the same as undefined. I would not give special treatment to `0`. 
2) You set the `min` for the field to `0`. I wouldn't make that restriction.

Reply:
Alright. Now I got it. 
yes both points are good considerations.
Thanks for the feedback!

by the way:  I didn’t find the unit tests for this class. Can I create one or did I just miss it somewhere?

Reply:
I am not so sure what the overall test strategy is. I generally see very few tests.

Reply:
Really have no clue to create valid vitests and running them.
If you could show me how I would like to add tests if needed.